File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import com.pinyougou.pojo.entity.PageResult;

import java.util.List;

/**
 * 品牌接口
 * @author Administrator
 *
 */
public interface BrandService {

	/**
	 * 测试方法
	 * @return
	 */
	List<TbBrand> findAll();

	/**
	 * 分页查询品牌
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	PageResult findPage(int pageNum,int pageSize);

	/**
	 * 增加品牌
	 * @param brand
	 */
	void add(TbBrand brand);

	/**
	 * 修改
	 * @param brand
	 */
	void update(TbBrand brand);

	/**
	 * 根据ID查找品牌
	 * @param id
	 * @return
	 */
	TbBrand selectOneById(Long id);

	/**
	 * 批量删除
	 * @param ids
	 */
	void delete(Long[] ids);

	/**
	 * 条件查询
	 * @param brand
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	PageResult findPage(TbBrand brand,int pageNum,int pageSize);
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {

    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    }

    this.findPage=function (page,rows) {
        return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
    }

    this.add=function (entity) {
        return $http.post('../brand/add.do',entity);
    }
    this.update=function (entity) {
        return $http.post('../brand/update.do',entity);
    }
    this.delete=function (ids) {
        return $http.get('../brand/delete.do?ids='+ids);
    }
    this.selectById=function (id) {
        return $http.get('../brand/selectOneById.do?id='+id);
    }
    this.search=function (page,rows,serachEntity) {
        return $http.post('../brand/search.do?page='+page+'&rows='+rows,serachEntity);
    }
});